Frameworks/Foundation/NSUserDefaults.mm

```diff
@@ -119,16 +132,44 @@ - (void)setPersistentDomain:(id)domain forName:(NSString*)name {
  @Status Interoperable
 */
 - (BOOL)synchronize {
-    return CFPreferencesAppSynchronize(kCFPreferencesCurrentApplication);
+    _cacheLock.lock();
```

Comment:
ok then let's do it everywhere.

---
In reply to: [70694534](https://github.com/Microsoft/WinObjC/pull/612#discussion_r70694534) [](ancestors = 70694534,70674087,70540876)